lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -105,49 +188,589 @@ TxOut
+                    pure (checkpointFromEntity cp utxo ins outs <$> s)
+                Nothing -> pure Nothing
+
+        {-----------------------------------------------------------------------
+                                   Wallet Metadata
+        -----------------------------------------------------------------------}
+
+        , putWalletMeta = \(PrimaryKey wid) meta ->
+            ExceptT $ unsafeRunQuery conn $
+            selectWallet wid >>= \case
+                Just _ -> do
+                    updateWhere [WalTableId ==. wid]
+                        (mkWalletMetadataUpdate meta)
+                    pure $ Right ()
+                Nothing -> pure $ Left $ ErrNoSuchWallet wid
+
+        , readWalletMeta = \(PrimaryKey wid) -> unsafeRunQuery conn $
+               fmap (metadataFromEntity . entityVal) <$> selectFirst [WalTableId ==. wid] []
+
+        {-----------------------------------------------------------------------
+                                     Tx History
+        -----------------------------------------------------------------------}
+
+        , putTxHistory = \(PrimaryKey wid) txs ->
+            ExceptT $ unsafeRunQuery conn $
+            selectWallet wid >>= \case
+                Just _ -> do
+                    let (metas, txins, txouts) = mkTxHistory wid txs
+                    putMany metas
+                    putMany txins
+                    putMany txouts
+                    pure $ Right ()
+                Nothing -> pure $ Left $ ErrNoSuchWallet wid
+
+        , readTxHistory = \(PrimaryKey wid) -> unsafeRunQuery conn $
+            selectTxHistory wid
+
+        {-----------------------------------------------------------------------
+                                       Keystore
+        -----------------------------------------------------------------------}
+
+        , putPrivateKey = \(PrimaryKey wid) key ->
+            ExceptT $ unsafeRunQuery conn $
+            selectWallet wid >>= \case
+                Just _ -> Right <$> do
+                    deleteWhere [PrivateKeyTableWalletId ==. wid]
+                    insert_ (mkPrivateKeyEntity wid key)
+                Nothing -> pure $ Left $ ErrNoSuchWallet wid
+
+        , readPrivateKey = \(PrimaryKey wid) ->
+            unsafeRunQuery conn $ let
+                keys = selectFirst [PrivateKeyTableWalletId ==. wid] []
+                toMaybe = either (const Nothing) Just
+            in (>>= toMaybe . privateKeyFromEntity . entityVal) <$> keys
+
+        {-----------------------------------------------------------------------
+                                       Lock
+        -----------------------------------------------------------------------}
+
+        , withLock = \_action -> error "withLock to be implemented"
+
+        }
+
+----------------------------------------------------------------------------
+-- Conversion between Persistent table types and wallet types
```

Comment:
It will be shorter and neater without name prefixes. All these table and column types are local to the Sqlite module. The actual wallet types are qualified with W.